capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+# Copyright 2020 Arista Networks. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+""" arista traffic-policy generator."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import datetime
+import copy
+
+import six
+from absl import logging
+from capirca.lib import aclgenerator
+
+#          1         2         3
+# 123456789012345678901234567890123456789
+# traffic-policies
+#    traffic-policy foo
+#      match dos-attaqrs-source-ip ipv4    << TERM_INDENT
+#         !! i am a comment, hear me rawr  << MATCH_INDENT
+#         source prefix field-set          << MATCH_INDENT
+#         !
+#         actions
+#            counter edge-dos-attaqrs-source-ip-count  << ACTION_INDENT
+#            drop
+#      !
+#
+#          1         2         3
+# 123456789012345678901234567890123456789
+# traffic-policies
+#    field-set ipv4 prefix dst-hjjqurby6yftqk6fa3xx4fas << TERM_INDENT
+#       0.0.0.0/0                                       << MATCH_INDENT
+#       except 34.64.0.0/26
+#    !
+#    field-set ipv4 prefix dst-hjjqurby6yftqk6fa3xx4fas
+
+# various indentation constants - see above
+INDENT_STR = " " * 3  # 3 spaces
+TERM_INDENT = 2 * INDENT_STR
+MATCH_INDENT = 3 * INDENT_STR
+ACTION_INDENT = 4 * INDENT_STR
+
+DEFAULT_TERM_SUFFIX = "-all-default"
+
+
+# generic error class
+class Error(Exception):
+    pass
+
+
+class TcpEstablishedWithNonTcpError(Error):
+    pass
+
+
+class AristaTpDuplicateTermError(Error):
```

Comment:
Please reuse existing error classes wherever possible. We already have a DuplicateTermError in aclgenerator.py:
https://github.com/google/capirca/blob/188fd110dbf4358f4f46a730d18f49ccc1981e53/capirca/lib/aclgenerator.py#L58